File: examples/java/Iteration.java
package main;

import java.util.List;
import java.util.LinkedList;

public class Iteration{
	static final List<String> values = new LinkedList<String>();

	static {
		values.add("hello");
		values.add("tech");
		values.add("wednesday");
	}

	public static void main(String[] args) {
		for (int i = 0, len = values.size(); i < len; i++) { // HL
			System.out.println(values.get(i));
		} // HL

		for (final String x : values) { // HL
			System.out.println(x);
		} // HL
	}
}

File: examples/javascript/iteration.js
'use strict';

var lg = console.log.bind(console);
var arr = ['hello', 'tech', 'wednesday'];

for (var i = 0, len = arr.length; i < len; i++) {
    lg(arr[i]);
}

arr.forEach(function (element) { lg(element); } );

arr.forEach(element => lg(element));

Array.prototype.forEach.call(arr, function (element) { lg(element); });

Array.prototype.forEach.call(arr, element => { lg(element); });

for (let element of arr) { lg(element); }
